app/javascript/components/WorkForm.jsx

```diff
@@ -0,0 +1,201 @@
+import PropTypes from "prop-types"
+import React from 'react';
+import { Button, Dialog, Intent } from "@blueprintjs/core"
+import { getCSRFToken } from '../shared/helpers/form_helpers.js'
+import Dropzone from "react-dropzone";
+import Dropdown from 'react-dropdown';
+import 'react-dropdown/style.css';
+
+class WorkForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      work: {
+        title: this.props.title,
+        material: this.props.material,
+        medium: this.props.medium,
+        availability: this.props.availability,
+        images: this.props.images,
+        artist_id: this.props.artist_id,
+      },
+      route: this.props.route,
+      method: this.props.method,
+      categories: {
+        "medium": 0,
+        "availability": 0
+      }
+    }
+  }
+
+  componentDidMount = () => {
+    const route = APIRoutes.works.categories;
+    Requester.get(route).then(
+      response => {
+        this.setState({
+          categories: response,
+          componentDidMount: true
+        });
+      },
+      error => {
+        console.error(error);
+      }
+    );
+  }
+
+  handleChange = (event) => {
+    const work = this.state.work;
+    work[event.target.name] = event.target.value;
+    this.setState({ work: work });
+  }
+
+  handleFormChange = (formAttr, value) => {
+    const work = this.state.work;
+    work[formAttr] = value;
+    this.setState({ work: work });
+  }
+
+  handleSubmit = (event) => {
+
+    event.preventDefault();
+
+    let formData = new FormData();
+    formData.append('work[title]', this.state.work.title);
+    formData.append('work[material]', this.state.work.material);
+    formData.append('work[medium]', this.state.work.medium);
+    formData.append('work[availability]', this.state.work.availability);
+    formData.append('work[artist_id]', this.state.work.artist_id);
+
+    this.state.work.images.forEach((img) => {
+      formData.append('work[attachments_attributes][]', img);
+    });
+
+    fetch(this.state.route, {
+      method: this.state.method,
+      body: formData,
+      credentials: 'same-origin',
+      headers: {
+        "X_CSRF-Token": document.getElementsByName("csrf-token")[0].content
+      }
+    }).then((data) => {
+      window.location = `/artists/` + this.state.work.artist_id;
+    }).catch((data) => {
+      console.error(data);
+    });
+  }
+
+  onDrop = (images) => {
+    var currentImages = this.state.work.images;
+    const [newImages] = images;
+    currentImages.push(newImages);
+    console.log(currentImages);
+
+    // Assign files dropped into component into state
+    var work = this.state.work;
+    work.images = currentImages;
+    this.setState({ work: work });
+  }
+
+  selectFeaturedImage = (img) => {
+    var work = this.state.work;
```

Comment:
typically let is used instead of var for variables, and const is used for variables that are constants